clients/conditionalaccesspolicy.go

```diff
@@ -0,0 +1,147 @@
+	params := url.Values{}
+	if filter != "" {
+		params.Add("$filter", filter)
+	}
+	resp, status, _, err := c.BaseClient.Get(ctx, base.GetHttpRequestInput{
+		ValidStatusCodes: []int{http.StatusOK},
+		Uri: base.Uri{
+			Entity:      "/identity/conditionalAccess/policies",
+			Params:      params,
+			HasTenantId: true,
+		},
+	})
+	if err != nil {
+		return nil, status, err
+	}
+	defer resp.Body.Close()
+	respBody, _ := ioutil.ReadAll(resp.Body)
+	var data struct {
+		ConditionalAccessPolicys []models.ConditionalAccessPolicy `json:"value"`
+	}
+	if err := json.Unmarshal(respBody, &data); err != nil {
+		return nil, status, err
+	}
+	return &data.ConditionalAccessPolicys, status, nil
+}
+
+// Create creates a new ConditionalAccessPolicy.
+func (c *ConditionalAccessPolicyClient) Create(ctx context.Context, conditionalAccessPolicy models.ConditionalAccessPolicy) (*models.ConditionalAccessPolicy, int, error) {
+	var status int
+	body, err := json.Marshal(conditionalAccessPolicy)
+	if err != nil {
+		return nil, status, err
+	}
+	resp, status, _, err := c.BaseClient.Post(ctx, base.PostHttpRequestInput{
+		Body:             body,
+		ValidStatusCodes: []int{http.StatusCreated},
+		Uri: base.Uri{
+			Entity:      "/identity/conditionalAccess/policies",
+			HasTenantId: true,
+		},
+	})
+	if err != nil {
+		return nil, status, err
+	}
+	defer resp.Body.Close()
+	respBody, _ := ioutil.ReadAll(resp.Body)
+	var newConditionalAccessPolicy models.ConditionalAccessPolicy
+	if err := json.Unmarshal(respBody, &newConditionalAccessPolicy); err != nil {
+		return nil, status, err
+	}
+	return &newConditionalAccessPolicy, status, nil
+}
+
+// Get retrieves an ConditionalAccessPolicy.
+func (c *ConditionalAccessPolicyClient) Get(ctx context.Context, id string) (*models.ConditionalAccessPolicy, int, error) {
+	resp, status, _, err := c.BaseClient.Get(ctx, base.GetHttpRequestInput{
+		ValidStatusCodes: []int{http.StatusOK},
+		Uri: base.Uri{
+			Entity:      fmt.Sprintf("/identity/conditionalAccess/policies/%s", id),
+			HasTenantId: true,
+		},
+	})
+	if err != nil {
+		return nil, status, err
+	}
+	defer resp.Body.Close()
+	respBody, _ := ioutil.ReadAll(resp.Body)
+	var conditionalAccessPolicy models.ConditionalAccessPolicy
+	if err := json.Unmarshal(respBody, &conditionalAccessPolicy); err != nil {
+		return nil, status, err
+	}
+	return &conditionalAccessPolicy, status, nil
+}
+
+// Update amends an existing ConditionalAccessPolicy.
+func (c *ConditionalAccessPolicyClient) Update(ctx context.Context, conditionalAccessPolicy models.ConditionalAccessPolicy) (int, error) {
+	var status int
+	if conditionalAccessPolicy.ID == nil {
+		return status, errors.New("cannot update conditionalAccessPolicy with nil ID")
+	}
+
+	// This API does not handle PATCH on some properties
+	// i.e. cannot update values such as CreatedDateTime
+	conditionalAccessPolicy.CreatedDateTime = nil
+	conditionalAccessPolicy.ModifiedDateTime = nil
```

Comment:
@manicminer what are your thoughts on this?

The /identity/conditionalAccess/policies/%s endpoint does not allow these values to be passed to the PATCH request. 

I have test to protect against unwanted mutations. see line 76-78 in the test.go

Reply:
@liammoat I'm not sure this is needed when the `omitempty` flag is set in the struct tag? If you leave those fields as nil in the model when updating, they shouldn't be included in the body of the patch request.

If a user inadvertently specifies those when updating and receives an API error, I'd rather that were exposed than try to suppress it by nilling those fields.

Reply:
Thanks @manicminer. Indeed `omitempty` removes it from the JSON payload if nil. I had an issue where the UPDATE test was "reusing" the object returned by the GET request test.

I have removed the code to nil these fields, and instead create a new object for the PATCH request with only the required fields `id` and `displayname`

This works well. Thanks for you feedback